src/components/container/BloodPressureVisualization/BloodPressureVisualization.tsx

```diff
@@ -50,7 +50,12 @@ export default function (props: BloodPressureVisualizationProps) {
     const yInterval: ClosedInterval = { values: [_minDiastolic, _maxSystolic] };
     const axis: Axis = { yRange: yInterval, yIncrement: 50, xIncrement: (80 / 7) };
     const [bloodPressureData, setDataForGraph] = useState<Map<string, BloodPressureDumbbell> | undefined>(undefined);
-    const [datePagerStartDate, setStartOfWeek] = useState<Date>(startOfDay(getWeekStart(props.weekStartsOn ?? "Monday")));
+    const dateRangeContext = useContext(DateRangeContext);
+    let intervalStart = dateRangeContext?.intervalStart ?? startOfDay(getWeekStart(props.weekStartsOn ?? "Monday"));
+
+    if ( dateRangeContext && dateRangeContext?.intervalType !== "Week"){
```

Comment:
This check could probably move above the `let intervalStart ...` computation too.

Also, the `?` is not needed in the second condition because the first condition already establishes that it is defined.